src/kinetics/GasKinetics.cpp

```diff
@@ -87,6 +87,15 @@ void GasKinetics::update_rates_C()
         m_falloff_concm.update(m_phys_conc, ctot, concm_falloff_values.data());
     }
 
+    // Third-body objects interacting with MultiRate evaluator
+    if (!concm_multi_values.empty()) {
+        // using pre-existing third-body handlers requires copying;
+        m_multi_concm.update(m_phys_conc, ctot, concm_multi_values.data());
+        for (size_t i = 0; i < m_multi_indices.size(); i++) {
+            m_concm[m_multi_indices[i]] = concm_multi_values[i];
+        }
+    }
+
```

Comment:
This is an aspect where I mostly implemented things to minimize changes in interfaces, as I will revisit this in more detail once I get to `Falloff` reactions.

Reply:
I looked at this one more time since I first responded: my approach here was to handle `ThirdBody` separately from `ReactionRate`. In a follow-up PR, this would be used to hand off third-body concentrations to `FallOffRate` objects.

Reply:
Leaving a `@TODO` comment in `MultiRate.h`